crates/bevy_ecs/src/entity/clone_entities.rs

```diff
@@ -175,29 +185,51 @@ impl<'w> EntityCloneBuilder<'w> {
             filter_allows_components,
             filter: Arc::new(filter),
             clone_handlers_overrides: Arc::new(clone_handlers_overrides),
+            move_components,
         }
         .clone_entity(world);
 
         world.flush_commands();
     }
 
+    /// Allows for a scoped mode where any changes to the filter that allow/deny components
+    /// will also allow/deny those components' required components, recursively.
+    pub fn with_required_components(
```

Comment:
Very cool. @cart is almost certainly going to want to want this on by default, possibly without the ability to disable it. The vision is to treat these as "one indivisible object" wherever possible, at least for "required" components :)

I agree with that philosophy, and I think the toggle should basically be "replace required components with their default values". The default value should be "copy all required components that you can".

Reply:
A bit off-topic for this PR, but I wonder if `DynamicSceneBuilder` should also extract required components in that case?

Reply:
That's essentially what the toggle is, since components are ultimately inserted with normal `insert` methods. I can switch the default though.

Here's a couple colors for the bikeshed:
- `default_required_components`
- `with_default_required_components`
- `replace_required_components`
- `replace_required_components_with_default`

Also, if `move_on_clone` is on, should required components still be removed if they weren't cloned?

Reply:
Yeah, we should probably be unifying the mechanisms between this and the scenes stuff. Another PR / issue though! As for the bikeshed, go with the long and clear one please: this is quite niche :)